File: src/main/java/com/hotel/demo/service/IDocumentService.java
package com.hotel.demo.service;

import com.hotel.demo.dto.DocumentDto;
import com.hotel.demo.model.Document;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.math.BigDecimal;
import java.sql.SQLException;
import java.util.List;

public interface IDocumentService {

    DocumentDto createDocument(DocumentDto documentDto);

List<DocumentDto> getDocuments(String val);
}

File: src/main/java/com/hotel/demo/service/IDocumentService.kt
package com.hotel.demo.service

import com.hotel.demo.dto.DocumentDto
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface IDocumentService {

    fun createDocument(documentDto: DocumentDto): Mono<DocumentDto>
    fun getDocuments(value: String, key: String?): Flux<DocumentDto>
}
